src/pegout/store/FlyoverPegout/actions.ts

```diff
@@ -149,14 +147,27 @@ export const actions: ActionTree<FlyoverPegoutState, RootState> = {
             productFeeAmount: quote2wp.quote.productFeeAmount,
             value: quote2wp.quote.value,
           };
-          const differences = compareObjects(
-            reducedCurrentQuote as unknown as { [key: string]: unknown },
-            reducedNewQuote as unknown as { [key: string]: unknown },
-          );
-          if (differences.length === 0) {
+          const zeroWei = new WeiBig(0, 'wei');
+          const currentQuoteTotalFee = (reducedCurrentQuote.callFee ?? zeroWei)
+            .plus(reducedCurrentQuote.gasFee ?? zeroWei)
+            .plus(reducedCurrentQuote.penaltyFee ?? zeroWei)
+            .plus(reducedCurrentQuote.productFeeAmount ?? zeroWei)
+            .plus(reducedCurrentQuote.value ?? zeroWei);
+          const newQuoteTotalFee = (reducedNewQuote.callFee ?? zeroWei)
+            .plus(reducedNewQuote.gasFee ?? zeroWei)
+            .plus(reducedNewQuote.penaltyFee ?? zeroWei)
+            .plus(reducedNewQuote.productFeeAmount ?? zeroWei)
+            .plus(reducedNewQuote.value ?? zeroWei);
+          const largest = newQuoteTotalFee
+            .gt(currentQuoteTotalFee) ? newQuoteTotalFee : currentQuoteTotalFee;
+          const minor = newQuoteTotalFee
+            .gt(currentQuoteTotalFee) ? currentQuoteTotalFee : newQuoteTotalFee;
+          const difference = ((largest.minus(minor)).mul('100')).div(largest.toRBTCString());
+          if (+difference.toRBTCString() <= EnvironmentAccessorService.getEnvironmentVariables()
```

Comment:
Using `Number()` instead of `+` for type conversion could make the intention clearer. What do you think?

Reply:
ok 👌